packages/cosmic-proto/scripts/codegen.cjs

```diff
@@ -0,0 +1,101 @@
+const { join } = require('path');
+const process = require('process');
+const telescope = require('@cosmology/telescope').default;
+const rimraf = require('rimraf').rimrafSync;
+
+const protoDirs = [join(__dirname, '/../proto')];
+const outPath = join(__dirname, '../src/codegen');
+rimraf(outPath);
+
+telescope({
+  protoDirs,
+  outPath,
+  options: {
+    tsDisable: {
+      // FIXME types aren't resolving correctly
+      disableAll: true,
+      files: [
+        'cosmos/authz/v1beta1/tx.amino.ts',
+        'cosmos/staking/v1beta1/tx.amino.ts',
+      ],
+      patterns: ['**/*amino.ts', '**/*registry.ts'],
+    },
+    interfaces: {
+      enabled: true,
+      useUnionTypes: false,
+    },
+    prototypes: {
+      includePackageVar: false,
+      excluded: {
+        packages: [
+          'ibc.applications.fee.v1', // issue with parsing protos (LCD routes with nested objects in params)
+
+          'cosmos.app.v1alpha1',
+          'cosmos.app.v1beta1',
+          'cosmos.base.kv.v1beta1',
+          'cosmos.base.reflection.v1beta1',
+          'cosmos.base.snapshots.v1beta1',
+          'cosmos.base.store.v1beta1',
+          'cosmos.base.tendermint.v1beta1',
+          'cosmos.crisis.v1beta1',
+          'cosmos.evidence.v1beta1',
+          'cosmos.genutil.v1beta1',
+
+          'cosmos.autocli.v1',
+
+          'cosmos.msg.v1',
+          'cosmos.nft.v1beta1',
+          'cosmos.capability.v1beta1',
+          'cosmos.orm.v1alpha1',
+          'cosmos.orm.v1',
+          'cosmos.slashing.v1beta1',
+          'google.api',
+          'ibc.core.port.v1',
+          'ibc.core.types.v1',
+        ],
+      },
+      methods: {
+        fromJSON: true,
+        toJSON: true,
+        encode: true,
+        decode: true,
+        fromPartial: true,
+        toAmino: false,
+        fromAmino: false,
+        fromProto: true,
+        toProto: true,
+      },
+      parser: {
+        keepCase: false,
+      },
+      typingsFormat: {
+        duration: 'duration',
+        timestamp: 'date',
+        useExact: false,
+        useDeepPartial: false,
+        num64: 'bigint',
+        customTypes: {
+          useCosmosSDKDec: true,
+        },
+      },
+    },
+    aminoEncoding: {
+      // Necessary for getSigningAgoricClient
+      enabled: true,
+    },
+    lcdClients: {
+      enabled: false,
+    },
+    rpcClients: {
+      enabled: true,
+      camelCase: true,
```

Comment:
Is this the setting that required changes in `agoric-cli` for the returned swingset params?

Reply:
I suppose so. Are you opposed to it? To me the camelCase is what one would expect in a JS API.

Reply:
@samsiegart you've written the most client code. What's your opinion?

Reply:
I'd guess camelCase, but not sure where this shows up

Reply:
@michaelfig made a good point about sticking to defaults. I took this option out to adopt the defaults. Turns out `true` is the Telescope default

Reply:
@michaelfig made a good point about sticking to defaults. I took this option out to adopt the defaults. Turns out `true` is the Telescope default.

@samsiegart it shows up in 7c01cb0231970c89379060606a5c18c47a478104

Reply:
Oh yea, `camelCase` please 👍

Reply:
I found the option that determines `power_flag_fees` vs `powerFlagFees`: parser.keepCase.

Setting it to `keepCase: true` matches the older snake_case. Removing it uses the camelCase, so that is the default of Telescope. Given our desire to match the defaults I'm leaving it alone.